fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/DATA/MfdFmsDataNavaid.tsx

```diff
@@ -0,0 +1,508 @@
+      [1 << 4]: NAVAID_TYPE_STRINGS[5],
+      [1 << 5]: NAVAID_TYPE_STRINGS[2],
+      [1 << 7]: NAVAID_TYPE_STRINGS[6],
+    };
+
+    if (!navaid.type && isNdbNavaid(navaid)) {
+      return 'NDB';
+    }
+
+    return typeMapping[navaid.type] || 'N/A';
+  }
+
+  private getILSCategory(navaid: VhfNavaid): string {
+    const catStr = navaid.name?.match(/CAT\s+(I{1,3})/)?.[1];
+
+    switch (catStr) {
+      case 'I':
+        return '1';
+      case 'II':
+        return '2';
+      case 'III':
+        return '3';
+      default:
+        return 'N/A';
+    }
+  }
+
+  private async getLocBearing(navaid: VhfNavaid): Promise<number | null> {
+    if (navaid.type === 1 << 7) {
+      const ilsData = await this.db.backendDatabase.getILSs([navaid.ident]);
+      const selectedILS = await this.props.mfd.deduplicateFacilities(ilsData);
+
+      if (selectedILS) {
+        return selectedILS.locBearing;
+      }
+    }
+    return null;
+  }
+
+  private async fetchNavaid(ident: string | null): Promise<void> {
+    if (!ident) return;
+
+    try {
+      const selectedNavaid = await this.getSelectedNavaid(ident);
+      const locBearing = await this.getLocBearing(selectedNavaid as VhfNavaid);
+      const frequency = this.formatFrequency(selectedNavaid.frequency);
+
+      const navaidInfo = this.createNavaidInfo(selectedNavaid, locBearing, frequency);
+
+      this.setNavaidType(selectedNavaid);
+      this.navaidData.set(navaidInfo);
+    } catch (e) {
+      console.error(`[FMS/FPM] Error loading navaid: ${e}`);
+    }
+  }
+
+  private async getSelectedNavaid(ident: string): Promise<VhfNavaid | NdbNavaid> {
+    const navaids = await this.db.searchAllNavaid(ident);
+    const selectedNavaid = await this.props.mfd.deduplicateFacilities(navaids);
+
+    if (!selectedNavaid) {
+      throw new Error(`[FMS/FPM] Can't find navaid with ICAO '${ident}'`);
+    }
+
+    return selectedNavaid;
+  }
+
+  private formatFrequency(frequency: number): string {
+    // As per the FCOM, we should display the frequency with one decimal place if the last digit is 0.
+    return frequency.toFixed(2).endsWith('0') ? frequency.toFixed(1) : frequency.toFixed(2);
+  }
+
+  private createNavaidInfo(
+    selectedNavaid: VhfNavaid | NdbNavaid,
+    locBearing: number | null,
+    frequency: string,
+  ): NavaidData {
+    const isVhf = !isNdbNavaid(selectedNavaid);
+    return {
+      airportIdent: isVhf ? selectedNavaid.airportIdent ?? 'N/A' : 'N/A',
+      ident: selectedNavaid.ident ?? 'N/A',
+      class: this.getNavaidType(selectedNavaid as VhfNavaid) ?? 'N/A',
+      latLong: coordinateToString(selectedNavaid.location.lat, selectedNavaid.location.long, false) ?? 'N/A',
+      elevation: 'N/A',
+      stationDeclination: isVhf
+        ? (selectedNavaid as VhfNavaid).stationDeclination?.toFixed(1).replace('-', '') ?? 'N/A'
+        : 'N/A',
+      declinationCardinalDir: isVhf ? (selectedNavaid.stationDeclination > 0 ? 'E' : 'W') : 'N/A',
```

Comment:
```suggestion
      declinationCardinalDir: isVhf ? (selectedNavaid.stationDeclination >= 0 ? 'E' : 'W') : 'N/A',
```
Also should be T for true-referenced navaids.